# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxPlatformGtk.cpp
# Commit: fe7075f48cf0
# Full Hash: fe7075f48cf031d5686b3268a0e738581721b386
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2020-07-23 03:20:25
# Description:
#   Bug 1650748 - Prevent DMABUF usage with GLX. r=nical
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D84256
# ==============================================================================

diff -r 36b357364a21 -r fe7075f48cf0 gfx/thebes/gfxPlatformGtk.cpp
--- a/gfx/thebes/gfxPlatformGtk.cpp	Wed Jul 22 21:00:42 2020 +0000
+++ b/gfx/thebes/gfxPlatformGtk.cpp	Wed Jul 22 18:58:11 2020 +0000
@@ -81,12 +81,17 @@
   mIsX11Display = gfxPlatform::IsHeadless()
                       ? false
                       : GDK_IS_X11_DISPLAY(gdk_display_get_default());
+  if (XRE_IsParentProcess()) {
 #ifdef MOZ_X11
-  if (mIsX11Display && XRE_IsParentProcess() &&
-      mozilla::Preferences::GetBool("gfx.xrender.enabled")) {
-    gfxVars::SetUseXRender(true);
+    if (mIsX11Display && mozilla::Preferences::GetBool("gfx.xrender.enabled")) {
+      gfxVars::SetUseXRender(true);
+    }
+#endif
+
+    if (IsWaylandDisplay() || (mIsX11Display && PR_GetEnv("MOZ_X11_EGL"))) {
+      gfxVars::SetUseEGL(true);
+    }
   }
-#endif
 
   InitBackendPrefs(GetBackendPrefs());
 
@@ -725,10 +730,11 @@
 
 #ifdef MOZ_WAYLAND
 bool gfxPlatformGtk::UseDMABufTextures() {
-  return IsWaylandDisplay() && GetDMABufDevice()->IsDMABufTexturesEnabled();
+  return gfxVars::UseEGL() && GetDMABufDevice()->IsDMABufTexturesEnabled();
 }
 bool gfxPlatformGtk::UseDMABufVideoTextures() {
-  return (GetDMABufDevice()->IsDMABufVideoTexturesEnabled() ||
+  return gfxVars::UseEGL() &&
+         (GetDMABufDevice()->IsDMABufVideoTexturesEnabled() ||
           StaticPrefs::media_ffmpeg_vaapi_enabled());
 }
 bool gfxPlatformGtk::UseHardwareVideoDecoding() {